examples/plotting/plot_rectangle.py

```diff
@@ -0,0 +1,82 @@
+"""
+==========================================================
+Drawing a Rectangle on a map
+==========================================================
+
+In this example we are going to look at how we can plot rectangles on maps
+while describing the dimensions using different methods.
+"""
+import matplotlib.pyplot as plt
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.data.sample
+import sunpy.map
+
+################################################################################
+# We use the `~sunpy.map.GenericMap.draw_rectangle` method to draw the rectangles.
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+fig = plt.figure(figsize=(5, 5))
+
+################################################################################
+# The first option is to provide an input `~astropy.coordinates.SkyCoord` as a
+# (2,)-shaped tuple in the form of ((x1, x2), (y1, y2)).
+fig.add_subplot(111, projection=aia_map)
+coords = SkyCoord((-400, 200) * u.arcsec, (-300, 100) * u.arcsec, frame=aia_map.coordinate_frame)
+aia_map.draw_rectangle(
+    coords,
+    color="blue",
+    linestyle="-",
+)
+aia_map.plot()
+plt.show()
+
+################################################################################
+# The second option is to pass `~astropy.coordinates.SkyCoord` bottom_left and
+# `~astropy.coordinates.SkyCoord` top_right.
+fig.add_subplot(111, projection=aia_map)
+bottom_left = SkyCoord(-400 * u.arcsec, -300 * u.arcsec, frame=aia_map.coordinate_frame)
+top_right = SkyCoord(300 * u.arcsec, 200 * u.arcsec, frame=aia_map.coordinate_frame)
+aia_map.draw_rectangle(
+    bottom_left,
+    top_right=top_right,
+    color="green",
+    linestyle="--",
+)
+aia_map.plot()
+plt.show()
+
+################################################################################
+# The third option is to provide `~astropy.coordinates.SkyCoord` bottom_left,
+# `~astropy.units.Quantity` width and `~astropy.units.Quantity` height.
+fig.add_subplot(111, projection=aia_map)
+bottom_left = SkyCoord(-500 * u.arcsec, -200 * u.arcsec, frame=aia_map.coordinate_frame)
+width = 700 * u.arcsec
+height = 500 * u.arcsec
```

Comment:
I think this should be the same bottom left corner as the first two. We should show how to draw the same rectangle in several different ways. 

```suggestion
width = 700 * u.arcsec
height = 500 * u.arcsec
```